docs/theory/swe.md

```diff
@@ -0,0 +1,342 @@
+$$
+
+where $l_{ij}$ is the length of the face connecting cells $i$ and $j$.
+
+With these definitions, the shallow water equations in cell $i$ are
+
+$$
+\frac{\partial\mathbf{U}_i}{\partial t} + \sum_j\mathbf{F}_{ij} l_{ij} = \mathbf{S}_i, \label{6}
+$$
+
+where the index $j$ in each term of the sum refers to a neighboring cell of cell $i$.
+
+### Boundary conditions
+
+To incorporate boundary conditions, we partition the domain boundary $\Gamma$
+into disjoint line segments, each of which represents a _boundary face_
+$\Gamma_i$. Every cell $j$ that touches the boundary $\Gamma$ has at least one
+boundary face so that 
+
+$$
+\partial\Omega_j \bigcup \Gamma \neq \varnothing.
+$$
+
+Such a cell is a _boundary cell_. The boundary $\Gamma$ consists entirely of
+faces of boundary cells.
+
+In dealing with boundary conditions, we must distinguish between the faces a
+boundary cell $i$ _does_ and _does not_ share with the boundary $\Gamma$:
+
+$$
+\frac{\partial\mathbf{U}_i}{\partial t} +
+\sum_{j: \partial\Omega_j\subset\Gamma}\mathbf{F}_{ij}^{\Gamma} l_{ij} +
+\sum_{j: \partial\Omega_j\not\subset\Gamma}\mathbf{F}_{ij} l_{ij}
+= \mathbf{S}_i. \label{7}
+$$
+
+To enforce boundary conditions, we must compute the effective boundary fluxes
+$\mathbf{F}_{ij}^{\Gamma}$ that appear in the first sum in $(\ref{7})$. These
+boundary fluxes have specific forms depending on their respective boundary
+conditions.
+
+### Source terms
+
+The term for the vector-valued source for the water's momentum, which we
+designate as $\vec{S}_{\vec{p}}$ is computed from the following integral:
+
+\begin{equation}
+\int_{d\Omega} \mathbf{\vec{S}}_\vec{p} d\Omega = \int_{\Omega} \left( -gh\nabla z - C_D \vec{u} \sqrt{u^2 + v^2} \right) d\Omega
+\end{equation}
+
+Here, $\nabla z = (\partial z/\partial x, \partial z/\partial y)$ is the
+two-dimensional gradient of the bed elevation function $z$ and $\vec{u} = (u, v)$
+is the flow velocity vector.
+
+The first of these terms, the _bed elevation slope_ term, represents the force
+of gravity on the water and can be approximated as
+
+\begin{equation}
+\int_{d\Omega_i} -gh\nabla z d\Omega_i \approx -gh\left(\overline{\frac{\partial z}{\partial x}}, \overline{\frac{\partial z}{\partial y}}\right) V_i
+\end{equation}
+
+where $V_i$ is the volume of cell $i$. For a triangular grid cell,
+
+\begin{eqnarray}
+\overline{\frac{\partial z}{\partial x}} &=& \frac{(y_2 - y_0)(z_1 - z_0) - (y_1 - y_0)(z_2 - z_0)}{(y_2 - y_0)(x_1 - x_0) - (y_1 - y_0)(x_2 - x_0)} \\
+\overline{\frac{\partial z}{\partial y}} &=& \frac{(x_2 - x_0)(z_1 - z_0) - (x_1 - x_0)(z_2 - z_0)}{(x_2 - y_0)(x_1 - x_0) - (x_1 - y_0)(y_2 - y_0)}.
+\end{eqnarray}
+
+The second term, the _roughness term_, is associated with bed friction:
+
+\begin{equation}
+\int_{\Omega_i} C_D \vec{u} \sqrt{u^2 + v^2} d\Omega_i \approx C_D \vec{u} \sqrt{u^2 + v^2} V_i
+\end{equation}
+
+### Evaluating normal fluxes
```

Comment:
How about moving this subsection before the subsection on 'Source terms' so that the computation of LHS terms is described before moving onto the description of RHS terms?